### DISCUSSION TITLE
Is --hosts-dir flag working in SOCI push command?

### DISCUSSION THREAD

**Discussion by ningziwen** [awslabs/soci-snapshotter#839] [Q&A] [answered]
I found `--hosts-dir` is part of [RegistryFlags](https://github.com/awslabs/soci-snapshotter/blob/f556b45e8e241ff243776e15352a6b7bd38f2161/cmd/soci/commands/push.go#L54C17-L54C17) being exposed in `soci push`, but it seems not working for me when I pass it. I tried to trace the code but didn't find any reference of using it in code.

If it is not supported yet and could be added later, we could open an issue to track it.

**Comment by Kern--** [comment, marked as answer]
The SOCO CLI flags are taken directly from `ctr` but only the most important are actually wired through. I think relying on `ctr` for code is a bad idea, so we should probably take the flags that are useful in the SOCI context and more them into the CLI directly.
